code/core/src/main/java/com/adobe/marketing/mobile/internal/eventhub/history/EventExtension.kt

```diff
@@ -0,0 +1,42 @@
+/*
+  Copyright 2025 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.internal.eventhub.history
+
+import com.adobe.marketing.mobile.Event
+import com.adobe.marketing.mobile.EventHistoryRequest
+import com.adobe.marketing.mobile.internal.util.flattening
+
+/**
+ * Creates an [EventHistoryRequest] from this event.
+ *
+ * @param from the start time that represents the lower bounds of the date range used when looking up an Event
+ * @param to the end time that represents the upper bounds of the date range used when looking up an Event
+ * @return an [EventHistoryRequest] with mask derived from this event's data
+ */
+internal fun Event.toEventHistoryRequest(
```

Comment:
Can this be moved to `com.adobe.marketing.mobile.internal.util` as `EventExtensions` is this only being used internally.